tools/src/main/java/org/apache/kafka/tools/VerifiableProducer.java

```diff
@@ -242,66 +207,174 @@ public String getValue(long val) {
     /** Close the producer to flush any remaining messages. */
     public void close() {
         producer.close();
-        System.out.println(shutdownString());
+        printJson(new ShutdownComplete());
     }
 
-    String shutdownString() {
-        Map<String, Object> data = new HashMap<>();
-        data.put("name", "shutdown_complete");
-        return toJsonString(data);
+    @JsonPropertyOrder({ "timestamp", "name" })
```

Comment:
Scanning through the diff my impression was that most of the additions were due to all the new POJOs. They don't really add value over just creating a map since we don't ever really use the objects other than for serialization. I think sticking to POJOs makes sense when you then have other code that takes advantage of the objects, gets proper type checking, etc. It just seems verbose here to me.

`createFromArgs` changes seem reasonable, I just wasn't sure about about the verbosity of the `ProducerEvent` approach.